client/allocrunner/consul_grpc_sock_hook.go

```diff
@@ -313,7 +313,8 @@ func proxyConn(ctx context.Context, logger hclog.Logger, destAddr string, conn n
 		defer cancel()
 		n, err := io.Copy(dest, conn)
 		if ctx.Err() == nil && err != nil {
-			logger.Warn("error proxying to Consul", "error", err, "dest", destAddr,
+			// expect disconnects when proxying http
+			logger.Trace("error proxying to Consul", "error", err, "dest", destAddr,
```

Comment:
My initial comment (https://github.com/hashicorp/nomad/issues/10804#issuecomment-885793340) also had logs about this in the other direction (namely consul -> socket). Shouldn't we downgrade the warning on line 335 as well?

Reply:
Good catch!